frontend/src/utils/methods.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ *  Removes an item if it matches the predicate, remove it if it doesn't.
+ *
+ * @param array     The original array
+ * @param item      The item to either remove or add to the array
+ * @param predicate Optional predicate to decide when to remove or add
+ * @returns The new, updated, array
+ */
+export const toggleFromArray: <T>(
+  array: T[],
+  item: T,
+  predicate?: (value: T, index: number, array: T[]) => boolean
+) => void = (array, item, predicate = (value) => value !== item) =>
+  array.includes(item) ? array.filter(predicate) : [...array, item];
```

Comment:
Hmm, this is a bit backwards on my head. Wouldn't the predicate return true only for the item to toggle? But here it's filtering to only return an array with the item(s) to be toggled

Reply:
The default predicate as it is now would return true for any item that's not the item to be filtered out. So the logic should work as intended - if the item is present then it returns a list without it (filtered), and if not it is added with the destructuring.  
If this is backwards, do you have any other suggestions for how to remove the item? Immediately I could do it with slice, but then I'd have to find the id, slice and after that return what remains (more code). But if that's more readable I could change it

Reply:
Updated now to invert the predicate, thoughts on the two approaches?